transifex/native/django/utils/gettext.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import unicode_literals
+
+import re
+
+VAR_FORMAT = 'variable_{cnt}'
+
+
+def source_string_to_native(string):
+    """Transform any gettext placeholders in the given string to Transifex
+    Native placeholders.
+
+    Usage:
+    >>> source_string_to_native('This is %s & %s')
+    <<< ('This is {variable_1} & {variable_2}', '['variable_1', 'variable_2']')
+
+    >>> source_string_to_native('This is %(foo)s and %(bar)s')
+    <<< ('This is {foo} and {bar}', ['foo', 'bar'])
+
+    :param unicode string: the source string
+    :return: a tuple, containing the new string (with Transifex Native syntax)
+        and a list of the names of all variables
+    :rtype: Tuple[unicode, list]
+    """
+    obj = {'cnt': 1, 'variables': []}  # Python 2 nonlocal workaround
+
+    def replace_named(match):
```

Comment:
I didn't find a PEP or any other best practice on this and all the examples I see (even official) don't use an underscore. I assume it's because you don't have access to the inner function from outside anyway, so there is no need to declare it as private.

Reply:
@dbendilas that's fine. The function underscoring is a personal thought to make the code easier to be followed.